blowfish.h

```diff
@@ -23,32 +42,36 @@ struct Blowfish_Info : public FixedBlockSize<8>, public VariableKeyLength<16, 4,
 /// \since Crypto++ 1.0
 class Blowfish : public Blowfish_Info, public BlockCipherDocumentation
 {
-	/// \brief Class specific implementation and overrides used to operate the cipher.
-	/// \details Implementations and overrides in \p Base apply to both \p ENCRYPTION and \p DECRYPTION directions
-	class CRYPTOPP_NO_VTABLE Base : public BlockCipherImpl<Blowfish_Info>
-	{
-	public:
-		void ProcessAndXorBlock(const byte *inBlock, const byte *xorBlock, byte *outBlock) const;
-		void UncheckedSetKey(const byte *key_string, unsigned int keylength, const NameValuePairs &params);
 
-	private:
-		void crypt_block(const word32 in[2], word32 out[2]) const;
+public:
+	typedef BlockCipherFinal<ENCRYPTION, Blowfish_Base<Blowfish_Info, BigEndian> > Encryption;
+	typedef BlockCipherFinal<DECRYPTION, Blowfish_Base<Blowfish_Info, BigEndian> > Decryption;
+};
+
+typedef Blowfish::Encryption BlowfishEncryption;
+typedef Blowfish::Decryption BlowfishDecryption;
 
-		static const word32 p_init[ROUNDS+2];
-		static const word32 s_init[4*256];
+/// \brief BlowfishCompat block cipher information
+struct BlowfishCompat_Info : public FixedBlockSize<8>, public VariableKeyLength<16, 4, 56>, public FixedRounds<16>
+{
+	CRYPTOPP_STATIC_CONSTEXPR const char* StaticAlgorithmName() {return "BlowfishCompat";}
+};
 
-		FixedSizeSecBlock<word32, ROUNDS+2> pbox;
-		FixedSizeSecBlock<word32, 4*256> sbox;
-	};
+/// \brief BlowfishCompat block cipher
+class BlowfishCompat : public BlowfishCompat_Info, public BlockCipherDocumentation

```

Comment:
Should this maybe become a `struct`, since everything in it is public?

Reply:
I've changed it to `struct` and updated the pull request - it looks better that way.